src/components/Asset/AssetActions/Trade/Swap.tsx

```diff
@@ -56,148 +61,189 @@ export default function Swap({
-      //     : amountOcean
-
-      // setMaximumDt(maximumDt.toString())
-      // setMaximumOcean(maximumOcean.toString())
-
-      // setOceanItem((prevState) => ({
-      //   ...prevState,
-      //   amount: amountOcean.toString(),
-      //   maxAmount: maximumOcean.toString()
-      // }))
-
-      // setDtItem((prevState) => ({
-      //   ...prevState,
-      //   amount: amountDataToken.toString(),
-      //   maxAmount: maximumDt.toString()
-      // }))
+          : new Decimal(maxBaseToken)
+
+      try {
+        const swapFee = await poolInstance.getSwapFee(
+          assetExtended.accessDetails?.addressOrId
+        )
+        setSwapFee(swapFee)
+
+        const maxBuyBaseToken = await poolInstance.getAmountOutExactIn(
+          assetExtended.accessDetails.addressOrId,
+          poolInfo.datatokenAddress,
+          poolInfo.baseTokenAddress,
+          amountDataToken.toString(),
+          swapFee
+        )
+        const maxBuyDt = await poolInstance.getAmountOutExactIn(
+          assetExtended.accessDetails?.addressOrId,
+          poolInfo.baseTokenAddress,
+          poolInfo.datatokenAddress,
+          amountBaseToken.toString(),
+          swapFee
+        )
+
+        const maximumDt =
+          values.type === 'buy'
+            ? amountDataToken.greaterThan(new Decimal(maxBuyDt))
+              ? maxBuyDt
+              : amountDataToken
+            : amountDataToken.greaterThan(new Decimal(balance.datatoken))
+            ? balance.datatoken
+            : amountDataToken
+
+        const maximumBaseToken =
+          values.type === 'sell'
+            ? amountBaseToken.greaterThan(new Decimal(maxBuyBaseToken))
+              ? maxBuyBaseToken
+              : amountBaseToken
+            : amountBaseToken.greaterThan(new Decimal(balance.baseToken))
+            ? balance.baseToken
+            : amountBaseToken
+
+        setMaximumDt(maximumDt.toString())
+        setMaximumBaseToken(maximumBaseToken.toString())
+
+        setBaseTokenItem((prevState) => ({
+          ...prevState,
+          amount: amountBaseToken.toString(),
+          maxAmount: maximumBaseToken.toString()
+        }))
+
+        setDtItem((prevState) => ({
+          ...prevState,
+          amount: amountDataToken.toString(),
+          maxAmount: maximumDt.toString()
+        }))
+      } catch (error) {
+        LoggerInstance.error(error.message)
+      }
     }
     calculateMaximum()
   }, [
-    asset,
-    maxOcean,
     maxDt,
+    maxBaseToken,
     balance,
-    values?.type,
+    values.type,
     setMaximumDt,
-    setMaximumOcean
+    setMaximumBaseToken,
+    assetExtended,
+    web3,
+    dtItem.token,
+    baseTokenItem.token
   ])
 
   const switchTokens = () => {
     setFieldValue('type', values.type === 'buy' ? 'sell' : 'buy')
-    setCoin(values.type === 'sell' ? 'OCEAN' : asset.datatokens[0].symbol)
+    setCoin(
+      values.type === 'sell' ? 'OCEAN' : assetExtended.datatokens[0].symbol
```

Comment:
last remaining hardcoded OCEAN in this PR, we should be able to replace all that with info from `usePool()` hook